src/core/vfm/metastore_test.go

```diff
@@ -0,0 +1,480 @@
+package vfm
```

Comment:
Yeah, the only reason I did subtests instead of separate methods like you suggested is because all of the tests use the same mongo client. I was unsure if it was possible to make some sort of init function which ran before all of the test functions. So, I decided to run all the tests as subtests from a single test method. This way I could initialize a client at the top of the test and then use it in all of the following subtests.

Reply:
This comment has the same response as another, so I'll just paste it here:

> You don't have to worry about it now, but a testing suite would've offered this functionality. Also, if you used a mock, this whole process would've been quite a bit more simple because there would've been no penalty for "disconnecting," since that would've been a NOP.

Reply:
In Go I think it's generally more common to use a fake, but either could work.